### ISSUE TITLE
kcom2: exited account ignored if it has no password set

### ISSUE THREAD

**Issue by Uafinder** [Kademi/kademi-dev#6171]
http://vladtest54eweb-vladtest54e.kademi-ci.co/vladtest54eweb-ecomstore/cart
use userqa02@mailinator.com
![image](https://user-images.githubusercontent.com/1945230/47069612-07567000-d1f8-11e8-943a-fba70a1a70a9.png)
![image](https://user-images.githubusercontent.com/1945230/47069621-0de4e780-d1f8-11e8-897d-c6e3627f029c.png)
http://vladtest54e.admin.kademi-ci.co/manageUsers/17947420/#summary-tab
so existed account ignored
and you allowed to create a duplicate account

**Comment by anhnt** [comment]
Yup, user with no password available will be considered as not found because they will not be able to login to system. and our system allows creating profiles with the same email.
Should be fine for me

**Comment by Uafinder** [comment]
ok, but we can't proceed with creating account because existed account
![image](https://user-images.githubusercontent.com/1945230/47070065-29042700-d1f9-11e8-9047-bf5169ad23ee.png)
http://vladtest54eweb-vladtest54e.kademi-ci.co/vladtest54eweb-ecomstore/cart
so either we need to update error or update existed account

@bradmac pls advice

**Comment by bradmac** [comment]
@anhnt please update so that when createAccount is called, we look for an existing profile by email address.

- If one is found, and it has no password, then update it and set password. 
- if a profile is found and it has a password, throw an error
- If no profile is found create an account normally

**Comment by anhnt** [comment]
@bradmac i noticed that `KademiSecurityManager.getPasswordManager` is not available in JS server side environment, looks like it's restricted so my code doesnt work

```
log.info('kcom2 passwordManager {}', securityManager.passwordManager);
                    var org = services.userManager.toOrg(orgData);
                    securityManager.passwordManager.setPassword(p, org, params.kcom2Password);
```

Please advise

**Comment by bradmac** [comment]
@anhnt PasswordManager isnt safe to be used from untrusted code. But i've added a setPassword method to UserManager which can be used.

```
public void setPassword(Profile user, String newPassword) throws PasswordInvalidException {
```

**Comment by anhnt** [comment]
Thanks it works
